crates/ruff/src/rules/flake8_bugbear/rules/unused_loop_control_variable.rs

```diff
@@ -154,20 +154,23 @@ pub(crate) fn unused_loop_control_variable(checker: &mut Checker, target: &Expr,
             },
             expr.range(),
         );
-        if let Some(rename) = rename {
-            if certainty.into() && checker.patch(diagnostic.kind.rule()) {
-                // Avoid fixing if the variable, or any future bindings to the variable, are
-                // used _after_ the loop.
-                let scope = checker.semantic().scope();
-                if scope
-                    .get_all(name)
-                    .map(|binding_id| checker.semantic().binding(binding_id))
-                    .all(|binding| !binding.is_used())
-                {
-                    diagnostic.set_fix(Fix::suggested(Edit::range_replacement(
-                        rename,
-                        expr.range(),
-                    )));
+        if checker.patch(diagnostic.kind.rule()) {
+            if let Some(rename) = rename {
+                if certainty.into() {
+                    // Avoid fixing if the variable, or any future bindings to the variable, are
+                    // used _after_ the loop.
+                    let scope = checker.semantic().scope();
+                    if scope
+                        .get_all(name)
+                        .map(|binding_id| checker.semantic().binding(binding_id))
+                        .filter(|binding| binding.range.start() >= expr.range().start())
```

Comment:
Was this just missing previously or... ?

Reply:
(I spent an embarrassing amount of time trying to figure out how this did the change described in the PR title before realizing it was a different rule)

Reply:
Yeah, sorry, the fix to `incorrect-dict-iterator` is based on the way this existing rule operates, and I noticed that they would both benefit from this condition. (We don't care about bindings that occur _before_ the loop.)